apps/condo/migrations/20230419122709-0261_ticket_feedbackadditionaloptions_and_more.js

```diff
@@ -0,0 +1,253 @@
+
+--
+-- Add field feedbackAdditionalOptions to ticket
+--
+ALTER TABLE "Ticket" ADD COLUMN "feedbackAdditionalOptions" jsonb NULL;
+--
+-- Add field feedbackComment to ticket
+--
+ALTER TABLE "Ticket" ADD COLUMN "feedbackComment" text NULL;
+--
+-- Add field feedbackUpdatedAt to ticket
+--
+ALTER TABLE "Ticket" ADD COLUMN "feedbackUpdatedAt" timestamp with time zone NULL;
+--
+-- Add field feedbackValue to ticket
+--
+ALTER TABLE "Ticket" ADD COLUMN "feedbackValue" varchar(50) NULL;
+--
+-- Add field feedbackAdditionalOptionsFrom to ticketchange
+--
+ALTER TABLE "TicketChange" ADD COLUMN "feedbackAdditionalOptionsFrom" jsonb NULL;
+--
+-- Add field feedbackAdditionalOptionsTo to ticketchange
+--
+ALTER TABLE "TicketChange" ADD COLUMN "feedbackAdditionalOptionsTo" jsonb NULL;
+--
+-- Add field feedbackCommentFrom to ticketchange
+--
+ALTER TABLE "TicketChange" ADD COLUMN "feedbackCommentFrom" text NULL;
+--
+-- Add field feedbackCommentTo to ticketchange
+--
+ALTER TABLE "TicketChange" ADD COLUMN "feedbackCommentTo" text NULL;
+--
+-- Add field feedbackValueFrom to ticketchange
+--
+ALTER TABLE "TicketChange" ADD COLUMN "feedbackValueFrom" varchar(50) NULL;
+--
+-- Add field feedbackValueTo to ticketchange
+--
+ALTER TABLE "TicketChange" ADD COLUMN "feedbackValueTo" varchar(50) NULL;
+--
+-- Add field feedbackAdditionalOptions to tickethistoryrecord
+--
+ALTER TABLE "TicketHistoryRecord" ADD COLUMN "feedbackAdditionalOptions" jsonb NULL;
+--
+-- Add field feedbackComment to tickethistoryrecord
+--
+ALTER TABLE "TicketHistoryRecord" ADD COLUMN "feedbackComment" text NULL;
+--
+-- Add field feedbackUpdatedAt to tickethistoryrecord
+--
+ALTER TABLE "TicketHistoryRecord" ADD COLUMN "feedbackUpdatedAt" timestamp with time zone NULL;
+--
+-- Add field feedbackValue to tickethistoryrecord
+--
+ALTER TABLE "TicketHistoryRecord" ADD COLUMN "feedbackValue" text NULL;
+
+--
+-- [CUSTOM] Add some helper functions
+--
+
+CREATE OR REPLACE FUNCTION getFeedbackOptionKeyFromReviewCommentOption(option text)
+    returns text
+as
+$getFeedbackOptionKeyFromReviewCommentOption$
+BEGIN
+    if option = 'Сделали на совесть' or option = 'Made in good faith' then return 'highQuality';
+    elseif option = 'Быстро сделали' or option = 'Quickly done' then return 'quickly';
+    elseif option = 'Сделали некачественно' or option = 'It was done poorly' then return 'lowQuality';
+    elseif option = 'Долго делали' or option = 'It took a long time' then return 'slowly';
+    else return '';
+    end if;
+end
+$getFeedbackOptionKeyFromReviewCommentOption$ LANGUAGE plpgsql;
+
+
+CREATE OR REPLACE FUNCTION convertReviewCommentToFeedbackAdditionalOptions(value text)
+    returns jsonb
+as
+$convertReviewCommentToFeedbackAdditionalOptions$
+declare
+    key1 text;
+    key2 text;
+BEGIN
+    key1 := getFeedbackOptionKeyFromReviewCommentOption(split_part(value, ';', 1));
+    key2 := getFeedbackOptionKeyFromReviewCommentOption(split_part(value, ';', 2));
```

Comment:
What if we have more than one options?

Reply:
We only have reviewComment in the format `{option1}` and `{option1};{option2}`. Also with `null` and ` ` (empty line) values.

All these cases are handled

Reply:
On prod there are only correct data, in right format